azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/DeployMojo.java

```
@@ -87,60 +92,68 @@ private IWebAppDeploymentSlot getDeploymentSlot(final WebAppConfig config) throw
                 .commit();
-        Log.info(String.format(CREATE_WEB_APP_DONE, result.entity().getName()));
+        Log.info(String.format(CREATE_WEB_APP_DONE, result.name()));
         return result;
     }
 
-    private IWebApp updateWebApp(final IWebApp webApp, final WebAppConfig webAppConfig) throws AzureExecutionException {
+    private IWebApp updateWebApp(final IWebApp webApp, final WebAppConfig webAppConfig) {
         // update app service plan
-        Log.info(String.format(UPDATE_WEBAPP, webApp.entity().getName()));
+        Log.info(String.format(UPDATE_WEBAPP, webApp.name()));
         final IAppServicePlan currentPlan = webApp.plan();
         IAppServicePlan targetServicePlan = StringUtils.isEmpty(webAppConfig.getServicePlanName()) ? currentPlan :
                 az.appServicePlan(getServicePlanResourceGroup(webAppConfig), webAppConfig.getServicePlanName());
         if (!targetServicePlan.exists()) {
             targetServicePlan = getOrCreateAppServicePlan(webAppConfig);
         }
         targetServicePlan.update().withPricingTier(webAppConfig.getPricingTier()).commit();
-        final IWebApp result = webApp.update().withPlan(targetServicePlan.entity().getId())
+        final IWebApp result = webApp.update().withPlan(targetServicePlan.id())
                 .withRuntime(webAppConfig.getRuntime())
                 .withDockerConfiguration(webAppConfig.getDockerConfiguration())
                 .withAppSettings(webAppConfig.getAppSettings())
                 .commit();
-        Log.info(String.format(UPDATE_WEBAPP_DONE, webApp.entity().getName()));
+        Log.info(String.format(UPDATE_WEBAPP_DONE, webApp.name()));
         return result;
     }
 
+    // todo: Extract resource group logic to library
+    private ResourceGroup getOrCreateResourceGroup(final WebAppConfig webAppConfig) {
+        try {
+            return az.getAzureResourceManager().resourceGroups().getByName(webAppConfig.getResourceGroup());
+        } catch (Exception e) {
```

identify the right exception when resource group doesn't exist, don't catch all Exceptions since some times it is due to other errors like Network issue.